tools/shared/pragma.c

```diff
@@ -658,7 +659,27 @@ do_sw(void)
           break;
         }
         LCASE(ctok);
-        if (strcmp(ctok, "always") == 0) {
+        if (strcmp(ctok, "vectorlength") == 0) {
+          assn( DIR_OFFSET( currdir, x[161]), 0 );
+          if( gtok() != T_LP){
+            return true;
+          }
+          while( true ){
+            typ = gtok();
+            if(typ==T_INT){
+              bset(DIR_OFFSET( currdir,x [27]),0x1);
+              assn(DIR_OFFSET( currdir, x[161]),(int)itok );
+              break;}
+            else if (strcmp(ctok,"fixed")==0){
+              bset(DIR_OFFSET(currdir,x [27]), 0x2 );
+              break;
+            }
+            else if (strcmp(ctok, "scalable") == 0 ){
+              bset(DIR_OFFSET(currdir, x [27]),0x4);
+              break;
+            }
+          }
```

Comment:
Yes, I think the sloppy handling of directive close parens is clearly a bit of a flang bug, but I don't think it is super important and also if we tighten it up we stand the risk of breaking existing (wrong) code. Even if we do fix it, we should do it in a different patch.

To be clear, I think we do need to parse the other problems correctly. Accepting garbage or non-comma separated lists should be fixed.

Reply:
in last patch warning is raised when vectorlength is used with not existing option eg: vectorlength(garbage)
-->fixed

Reply:
vectorlength(garbage) is handled as an error- an error is raised.
-->fixed

Reply:
LGTM, thanks for working to make it right. I think given the amount of code here, it would be worth lifting the parsing for vectorlength out into a separate static helper function in this file rather than being called inline here.